CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/base/BaseFragment.kt

```diff
@@ -0,0 +1,56 @@
+package co.nimblehq.coroutine.ui.base
+
+import android.os.Bundle
+import android.view.*
+import androidx.annotation.CallSuper
+import androidx.fragment.app.Fragment
+import co.nimblehq.coroutine.extension.hideSoftKeyboard
+import co.nimblehq.coroutine.ui.common.Toaster
+import co.nimblehq.coroutine.ui.userReadableMessage
+import javax.inject.Inject
+
+abstract class BaseFragment : Fragment(), BaseFragmentCallbacks {
+
+    @Inject
+    lateinit var toaster: Toaster
+
+    protected abstract val layoutRes: Int
+
+    @CallSuper
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (this as? BaseFragmentCallbacks)?.let { initViewModel() }
+    }
+
+    override fun initViewModel() {}
+
+    @CallSuper
+    override fun bindViewEvents() {
+        requireNotNull(view).setOnClickListener {
+            requireActivity().hideSoftKeyboard()
+        }
+    }
```

Comment:
Shall we move this below `onViewCreated(...)` ? 🙏

Reply:
Resolved 🙏🏼